### PR TITLE
Allows the HTTP server to use PEM truststore

### PR DESCRIPTION
**Pull request by cescoffier**
Previously, configuring the truststore with PEM files for mTLS (mutual TLS) usage was not supported, limiting options to JKS and P12 formats. This commit addresses this limitation by introducing support for PEM truststores in the HTTP server configuration.

Additionally, this commit accomplishes the following tasks:

- Removes hand-crafted certificates.
- Fixes issues related to alias passwords.
- Deprecates the usage of "key" properties, replacing them with "alias" properties for clarity and consistency.

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -374,7 +374,7 @@
             <dependency>
                 <groupId>me.escoffier.certs</groupId>
```
Comment (sberyozkin) [inline build-parent/pom.xml:375]: Can this project be hosted at quarkus.io, or smallrye ?

**Review by sberyozkin — APPROVED** [review]
Nice clean up, LGTM.

Have you already documented the `@Certificates` option ? I suppose may be it is better wait till the project is hosted somewhere in the Quarkus space ?

File: build-parent/pom.xml (hunk 1 above)
Comment (cescoffier) [reply build-parent/pom.xml:375]: Not for now. We will move it when I'm done with what I have in mind.
However, the benefits are so nice (seriously, not fearing expiration, making sure that it's done right, removing trust all..., I feel so much more comfortable) that I started adopting it in various places.

**Comment by cescoffier** [comment]
@sberyozkin the documentation is on https://github.com/cescoffier/certificate-generator until we move it.

File: build-parent/pom.xml (hunk 1 above)
Comment (sberyozkin) [reply build-parent/pom.xml:375]: @cescoffier Sounds good, thanks

**Comment by sberyozkin** [comment]
The BC JSSE test failure is likely related as it uses the cred provider

**Comment by cescoffier** [comment]
Yes, I had to change a few things in the credential provider support. I will have a look on Monday.

**Comment by cescoffier** [comment]
@sberyozkin it should be fixed now - was a bug in the new code.

**Pull request merged**